tests/common/test_cgroupapi.py

```diff
@@ -505,6 +497,88 @@ def mock_popen(*args, **kwargs):
+        # Mock systemd timeout and make sure the failure is only attributed to the extension if the command fails
+        # using the fallback option
+        original_popen = subprocess.Popen
+
+        def mock_popen(*args, **kwargs):
+            # Inject a syntax error to the call
+            if "systemd-run" in args[0]:
+                return
+            else:
+                return original_popen(*args, **kwargs)
+
+        expected_output = "[stdout]\n{0}\n\n\n[stderr]\n"
+
+        with tempfile.TemporaryFile(dir=self.tmp_dir, mode="w+b") as stdout:
+            with tempfile.TemporaryFile(dir=self.tmp_dir, mode="w+b") as stderr:
+                with patch("azurelinuxagent.common.cgroupapi.subprocess.Popen", side_effect=mock_popen):
+                    with patch("azurelinuxagent.common.cgroupapi.wait_for_process_completion_or_timeout",
+                               return_value=[True, None]):
+                        with patch("azurelinuxagent.common.cgroupapi.SystemdCgroupsApi.is_systemd_failure",
+                                   return_value=True):
+                            extension_cgroups, process_output = SystemdCgroupsApi().start_extension_command(
+                                extension_name="Microsoft.Compute.TestExtension-1.2.3",
+                                command="echo 'success'",
+                                timeout=300,
+                                shell=True,
+                                cwd=self.tmp_dir,
+                                env={},
+                                stdout=stdout,
+                                stderr=stderr)
+
+                            self.assertEquals(extension_cgroups, [])
+                            self.assertEquals(expected_output.format("success"), process_output)
+
+    @attr('requires_sudo')
+    @patch("azurelinuxagent.common.cgroupapi.add_event")
+    def test_start_extension_command_should_not_use_fallback_option_if_extension_fails(self, *args):
+        self.assertTrue(i_am_root(), "Test does not run when non-root")
+
+        with tempfile.TemporaryFile(dir=self.tmp_dir, mode="w+b") as stdout:
+            with tempfile.TemporaryFile(dir=self.tmp_dir, mode="w+b") as stderr:
+                with self.assertRaises(ExtensionError) as context_manager:
+                    SystemdCgroupsApi().start_extension_command(
+                        extension_name="Microsoft.Compute.TestExtension-1.2.3",
+                        command="ls folder_does_not_exist",
+                        timeout=300,
+                        shell=True,
+                        cwd=self.tmp_dir,
+                        env={},
+                        stdout=stdout,
+                        stderr=stderr)
+
+                    self.assertEquals(context_manager.exception.code, ExtensionErrorCodes.PluginUnknownFailure)
+                    self.assertIn("Non-zero exit code", ustr(context_manager.exception))
```

Comment:
how do we verify the fallback was not used?

Reply:
Added asserts for that